sql.go

```diff
@@ -133,9 +133,9 @@ func (s *sql) UpdatePrefix(prefix Prefix) (Prefix, error) {
 	if rows == 0 {
 		err := tx.Rollback()
 		if err != nil {
-			return Prefix{}, newOptimisticLockError("select for update did not effect any row, but rollback did not work:" + err.Error())
+			return Prefix{}, fmt.Errorf("select for update did not effect any row, but rollback did not work:%w", err)
```

Comment:
This changes the contract, because now the error is no longer the OptimisticLockError but the Error seen at Rollback. The client should really see the OptimisticLockError in my opinion.
We could even remove the rollback since if zero rows had been are affected this makes no sense to rollback "no changes".

Reply:
Yes thats right.

Reply:
still do the rollback, but error handling is removed.